content/post/next_compiler.md

```diff
@@ -0,0 +1,207 @@
+human behavior.
+
+Regarding source code, we are very lucky to **have such a dataset, and it is tremendously huge**: all
+the open source projects. Will it be GitHub or anything "in the wild" - source{d}'s data engineers
+estimate the overall size in petabytes. Now let's combine both statements into
+
+#### The naturalness hypothesis:
+
+Software is a form of human communication; software corpora have similar statistical properties
+to natural language corpora; and these properties can be exploited to build better software
+engineering tools. 
+
+This hypothesis fully expresses what we think and even checked several times (
+[topic modeling](https://arxiv.org/abs/1704.00135),
+[identifier embeddings](https://blog.sourced.tech/post/id2vec/),
+[code prediction](https://github.com/src-d/code-completion)). It is taken from
+["A Survey of Machine Learning for Big Code and Naturalness"](https://arxiv.org/pdf/1709.06182.pdf)
+by Allamanis et.al. This paper is an excellent overview of what we have today in MLoSC, and
+we are confident that we are still in the very early stages. Particularly, the study of
+how the brain works while reading source code depending on experience level is fantastic.
+
+The idea to "invert" Machine Learning is floating in the air. Jeff Dean has given a great talk
+at NIPS 2017 about ["Machine Learning for Systems"](http://learningsys.org/nips17/assets/slides/dean-nips17.pdf).
+While it pretty much focuses on runtime program behavior and Google problems, it has a very clear
+message: let's apply Machine Learning to the technical domains nobody has considered yet.
+
+Let me finish the first part with a clarification. The naturalness hypothesis does not by any means
+imply that Machine Learning can replace human coders. We are very far from that. At the same time
+better tools lead to higher productivity and thus less need for human workforce. Just as it
+happened many times before in the history.
+
+## Feedback
+
+Programming languages evolve, as the natural ones do. The rules of a natural language reflect
+the norms which are adopted by the majority of the speakers. There is evidence that the complexity
+of a natural language decreases over time. If you compare Russian language from 19th century
+to the one in 21st, you will surprise how ridiculously complex is the former compared to the
+latter: more letters, more rules, more exceptions. Funnily enough, we do not know any programming
+language which does becomes simpler or easier to learn over time.
+
+There are several reasons we can imagine for that. While in a natural language words are either
+dropped or replaces, programming languages keep gaining new functions yet at the same time do
+not remove old ones and maintain backwards compatibility. The sad example of Python 3 clearly
+shows that revolutionary changes harm so much that they can split the community.
+
+Francesc, the hed of our Developer Relations, presented a curious example of Go language evolution
```

Comment:
s/hed of our Developer Relations/VP of Developer Relations at source{d}/

Reply:
Also, please add a link to my twitter https://twitter.com/francesc from my name.